### PR TITLE
Add warning message in sdk/cli to consume new llm tools

### PR DESCRIPTION
**Pull request by 16oeahr**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-devkit/promptflow/_cli/data/chat_flow/template/flow.dag.yaml.jinja2 [hunk 1]
```diff
@@ -14,17 +14,17 @@ outputs:
     is_chat_output: true
 nodes:
 - name: chat
-  type: llm
+  type: custom_llm
   source:
```
Comment (16oeahr) [inline src/promptflow-devkit/promptflow/_cli/data/chat_flow/template/flow.dag.yaml.jinja2:18]: need to wait for new tool pypi

File: src/promptflow-devkit/promptflow/_sdk/_load_functions.py [hunk 2]
```diff
@@ -81,7 +82,13 @@ def load_flow(
     :return: A Flow object
     :rtype: ~promptflow._sdk.entities._flows.Flow
     """
-    return Flow.load(source, **kwargs)
+    flow = Flow.load(source, **kwargs)
+    if check_legacy_llm_in_flow_nodes(flow._data):
+        logger.warning(
+            "Please upgrade to the latest version of promptflow-tools to consume new LLM tools. "
```
Comment (16oeahr) [inline src/promptflow-devkit/promptflow/_sdk/_load_functions.py:88]: need to wait for new tool pypi release

File: src/promptflow-core/promptflow/_utils/flow_utils.py [hunk 3]
```diff
@@ -20,6 +20,7 @@
     PROMPTY_EXTENSION,
 )
 from promptflow._core._errors import MetaFileNotFound, MetaFileReadError
+from promptflow._core.tool import ToolType
```
Comment (chjinche) [inline src/promptflow-core/promptflow/_utils/flow_utils.py:23]: https://microsoft.github.io/promptflow/reference/python-library-reference/promptflow-core/promptflow.contracts.tool.html?highlight=tooltype#promptflow.contracts.tool.ToolType

File: src/promptflow-devkit/promptflow/_cli/data/chat_flow/template/flow.dag.yaml.jinja2 [hunk 4]
```diff
@@ -14,17 +14,17 @@ outputs:
     is_chat_output: true
 nodes:
 - name: chat
-  type: llm
+  type: custom_llm
```
Comment (chjinche) [inline src/promptflow-devkit/promptflow/_cli/data/chat_flow/template/flow.dag.yaml.jinja2:17]: feel risky to modify now, which may make local user fail to run new created flow.

File: src/promptflow-devkit/promptflow/_cli/data/chat_flow/template/flow.dag.yaml.jinja2 [hunk 5]
```diff
@@ -14,17 +14,17 @@ outputs:
     is_chat_output: true
```
Comment (chjinche) [inline src/promptflow-devkit/promptflow/_cli/data/chat_flow/template/flow.dag.yaml.jinja2:14]: why not change other flow templates?

File: src/promptflow-devkit/promptflow/_sdk/_load_functions.py (hunk 2 above)
Comment (chjinche) [reply src/promptflow-devkit/promptflow/_sdk/_load_functions.py:88]: Should specify which tool version.

Comment (chjinche) [inline src/promptflow-devkit/promptflow/_sdk/_load_functions.py:88]: Besides upgrade local pf-tools version, I think there are other things people also need to do to dismiss the warning.

**Comment by chjinche** [comment]
No PR description?

**Pull request closed without merging**